### ISSUE TITLE
Issue with using React.createRef with DockingLayout

### ISSUE THREAD

**Issue by ItsNoHax** [HTMLElements/smart-webcomponents-react#1]
Hi,

I've been running into a weird issue where Typescript is complaining about my ref usage.

Reproduction code:
```js
// class attribute
private dockinglayout = React.createRef<DockingLayout>();
// in render() method
<DockingLayout ref={this.dockinglayout} id="layout" layout={this.layout} draggable></DockingLayout>
```

Error is:
```
Type 'RefObject<DockingLayout>' is not assignable to type 'string | (string & ((instance: Element | null) => void)) | (string & RefObject<Element>) | (RefObject<DockingLayout> & string) | ... 6 more ... | undefined'.
  Type 'RefObject<DockingLayout>' is not assignable to type '((instance: DockingLayout | null) => void) & RefObject<Element>'.
    Type 'RefObject<DockingLayout>' is not assignable to type '(instance: DockingLayout | null) => void'.
      Type 'RefObject<DockingLayout>' provides no match for the signature '(instance: DockingLayout | null): void'.ts(2322)
index.d.ts(143, 9): The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<DockingLayout> & Readonly<HTMLProps<Element> & DockingLayoutProps> & Readonly<...>'
```

I think the problem lies in the following line in dockinglayout.d.ts
```js
export declare class DockingLayout extends React.Component<React.HTMLProps<Element> & DockingLayoutProps, any> {
```
When I update the line to use React.HTMLAttributes instead of HTMLProps the error dissapears:

```js
export declare class DockingLayout extends React.Component<React.HTMLAttributes<Element> & DockingLayoutProps, any> {
```

Some extra googling links:
https://stackoverflow.com/questions/50627493/how-extend-react-types-to-support-html-attributes-as-props
https://gist.github.com/mfal/88fe927f2061d22425cc78fcec8d260f

**Comment by bmarkov** [comment]
Thank you for the feedback.

**Comment by jqwidgets** [comment]
The issue is resolved in smart-webcomponents-react@8.0.4
